File: exercicios em Python/q34.py
import os

print("\n\nEste programa ler a nota individual de n alunos e determina a nota máxima, a nota mínima, a média da turma, a quantidade de notas maiores do que a média e a quantidade de notas inferiores ao valor da média, e quantos alunos foram aprovados!")
n = int(input("\nDigite a quantidade de alunos pertencente a sala: "))
i = 1
media = 0
rep = 0
apr = 0
mm = 0
mM = 0
while i <= n:
        print("\nDigite a nota do ",i," aluno:")
        nota = int(input(""))
        media = media + nota
        if i == 1:
                notamx = nota
                notamn = nota
                
        elif notamx < nota:
                notamx = nota

        elif notamn > nota:
                notamn = nota

        if nota < 7:
                rep = rep + 1

        elif nota >= 7:
                apr = apr + 1

        i = i + 1

print("\nAgora porfavor digite as notas do aluno novamente")
media = media / n
i = 1
while i <= n:
        print("\nDigite a nota do ",i," aluno:")
        nota = int(input(""))
        if nota < media:
                mm = mm + 1

        elif nota >= media:
                mM = mM + 1

        i = i + 1

print("\nA nota maxima foi ",notamx," e a nota minima ",notamn,". A media da turma foi ",media,", a quantidade de alunos assima da media da turma foi de ",mM," e a quantidade de menores foi ",mm,". E por fim a quantidade de alunos aprovados foram de ",apr," e de reprovados de ",rep,".")
print("\nObrigado por utilizar nosso sistema!")
os.system("pause")

File: exercicios em C/q34.cpp
#include <stdio.h>

int main(void){
	int i, nota, n, notaMax, notaMin, media, maiorM, menorM, alunosRep, alunosAprov;
	maiorM = 0; 
	menorM = 0;
	alunosRep = 0; 
	alunosAprov = 0;
	printf("\n\nEste programa ler notas variadas entre 0 e 10 de n alunos e determina, nota maxima, nota mminima, a media da turma, a quantidade de notas maiores do que a media, a quantidade de notas menores que a media, a quantidade de alunos aprovados, e reprovados.");
	printf("\n\nDigite a quantidade de alunos pertencente a turma: ");
	scanf("%d", &n);
	for(i = 1; i <= n; i++){
		printf("\n\nDigite a nota do %d aluno: ", i);
		scanf("%d", &nota);
		if (i == 1){
			notaMax = nota;
			notaMin = nota;
		}
		if(notaMax < nota){
			notaMax = nota;
		}
		else if(notaMin > nota){
			notaMin = nota;
		}
		if (nota < 7){
			alunosRep++;
		}
		else if(nota >= 7){
			alunosAprov++;
		}
	}
	printf("\n\nPorfavor Digite as notas do alunos novamente para o calculo da media");
	for(i = 1; i <= n; i++){
		printf("\n\nDigite a nota do %d novamente: ", i);
		scanf("%d", &nota);
		media = media + nota;
	}
	media = media / n;
	if (nota >= media){
		maiorM = maiorM + 1;
		}
	else if (nota < media){
		menorM = menorM + 1;
	}
	printf("\n\nA nota maxima foi %d, e a nota minima foi %d .", notaMax, notaMin);
	printf("\n\nA media da turma foi %d, a quantidade de notas maiores da media foi %d, e a quantidade menores foram %d", media, maiorM, menorM);
	printf("\n\nA quantidade de alunos aprovados foram %d, e a quantidade de reprovados foram %d.", alunosAprov, alunosRep);
	printf("\n\nObrigado utilizar nosso programa!");
	return 0;
}
